src/main/java/com/mraof/minestuck/entity/AnimatedCreatureEntity.java

```diff
@@ -0,0 +1,163 @@
+        return this.distanceToSqr(target) <= reach;
+    }
+
+    private void startAttack() {
+        if (heavyAttackTicks <= 0 && recoveryTicks <= 0) {
+            heavyAttackTicks = attackDelay;
+            setAttacking(true);
+        }
+    }
+
+    protected AnimationController<AnimatedCreatureEntity> createAnimation(String name, double speed, AnimationController.IAnimationPredicate<AnimatedCreatureEntity> predicate) {
+        AnimationController<AnimatedCreatureEntity> controller = new AnimationController<>(this, name, 0, predicate);
+        controller.setAnimationSpeed(speed);
+        return controller;
+    }
+
+    public AnimationFactory getFactory() {
+        return this.factory;
+    }
+
+    public void setAttackRecovery(int attackRecovery) {
+        this.attackRecovery = attackRecovery;
+    }
+
+    public void setAttackDelay(int attackDelay) {
+        this.attackDelay = attackDelay;
+    }
+
+    public void setAttacking(boolean attacking) {
+        if (attacking) {
+            this.entityData.set(CURRENT_ACTION, Actions.ATTACK.ordinal());
+        } else {
+            this.entityData.set(CURRENT_ACTION, Actions.NONE.ordinal());
+        }
+    }
+
+    public boolean isAttacking() {
+        return Actions.ATTACK.isEqual(this.entityData.get(CURRENT_ACTION));
+    }
+
+    public Actions getCurrentAction() {
+        return Actions.values()[this.entityData.get(CURRENT_ACTION)];
+    }
+
+    /**
+     * Used to set the entity's action
+     * @param action The action to be performed
+     * @param tickLength -1 to for infinite action or any positive value to set the tick length
+     */
+    public void setCurrentAction(Actions action, int tickLength) {
+        this.entityData.set(CURRENT_ACTION, action.ordinal());
+        this.actionResetTicks = tickLength;
+    }
+
+    /**
+     * Used to set the entity's action - will repeat indefinitely
+     * @param action The action to be performed
+     */
+    public void setCurrentAction(Actions action) {
+        setCurrentAction(action, -1);
+    }
+
+    public enum Actions {
+        NONE,
+        ATTACK,
+        TALK,
+        PANIC,
+        PANIC_RUN;
+
+        public boolean isEqual(int ordinal) {
+            return this.ordinal() == ordinal;
+        }
+    }
+
+    private static class DelayedAttackGoal extends MeleeAttackGoal {
+        private final AnimatedCreatureEntity entity;
+
+        public DelayedAttackGoal(AnimatedCreatureEntity entity, float speed, boolean useMemory) {
+            super(entity, speed, useMemory);
+            this.entity = entity;
+        }
+
+        @Override
+        protected void checkAndPerformAttack(LivingEntity enemy, double distToEnemySqr) {
+            double reach = this.getAttackReachSqr(enemy);
+            if (distToEnemySqr <= reach && this.ticksUntilNextAttack <= 0) {
+                this.resetAttackCooldown();
+                this.mob.swing(Hand.MAIN_HAND);
```

Comment:
Does swinging the hand accomplish anything? I could imagine some scenarios in which tracking hand swinging is valuable even if not rendered but was unsure if there was a solid example you already know of or used it with

Reply:
I am ok if you leave it in if you believe it has potential use. Its definitely harmless as far as can be seen now